### ISSUE TITLE
Release 0.8.0

### ISSUE THREAD

**Issue by jstuyts** [tatsuhiro-t/argparse4j#101]
There have been quite a number of changes, and I think together they would make a nice new release.

There are still 5 outstanding issues in 3 categories:
* More flexible help formatting:
    * https://github.com/tatsuhiro-t/argparse4j/issues/48 - Implement help formatter
    * https://github.com/tatsuhiro-t/argparse4j/issues/81 - Positional arguments should appear before optional arguments in help output
* Deprecation/migration support:
    * https://github.com/tatsuhiro-t/argparse4j/issues/28 - Check required options based off the destination, not the presence of the option name?
    * https://github.com/tatsuhiro-t/argparse4j/issues/44 - Provide way of transitioning from -argument to --argument
* Supporting a default sub parser:
    * https://github.com/tatsuhiro-t/argparse4j/issues/49 - Default subparser? 

Most of these issues are pretty old, and the first 2 categories require substantial changes and testing. So I think it is better to push these issues to a later release.

We do have a number of source incompatible changes, which also may lead to run time errors if argparse4j is extended in very specific ways: 3 methods added to 2 interfaces. I think the chances of run time problems are very slim. It needs to be documented in `NEWS` though.

As far as I can tell the only remaining task before it is release ready, is updating `NEWS`.

**Comment by jstuyts** [comment]
@tatsuhiro-t Do you agree 0.8.0 is ready for release? If so, when do you have time to create and upload it?

**Comment by tatsuhiro-t** [comment]
Agreed.  I can spare time for making a release.  It's been a while since the last release, so I have to re-learn about maven and nexus stuff.  How about the next weekend?

**Comment by jstuyts** [comment]
Next weekend sounds great.

**Comment by tatsuhiro-t** [comment]
I uploaded snapshots upload to OSSRH successfully.  All jars were there, so it looks good.
To make a release, I just remove "-SNAPSHOT" from all pom files, and then run deploy, and updates web pages.
@jstuyts are you OK for me to make a release as a current form of this repository?

**Comment by jstuyts** [comment]
@tatsuhiro-t  Yes, I think everything is ready. There are quite a number of usability improvements that users will appreciate. Documentation and news files have been updated.

The remaining open issues have a major impact, so are better postponed to a future release.

**Comment by tatsuhiro-t** [comment]
Thank you.  I do release work from now.

**Comment by tatsuhiro-t** [comment]
All release works have done.  Thank you!  Looking forward to 0.9.0 release.

**Comment by jstuyts** [comment]
It looks like your forget to publish the `argparse4j-root` POM. My build fails because it cannot find this POM in Maven Central.

**Comment by tatsuhiro-t** [comment]
I confirmed that argparse4j-root is missing as well as argparse4j-extensions.
When I deployed 0.8.0-SNAPSHOT staging build, they were uploaded, but they weren't with 0.8.0.
I suspect that we need org.sonatype.plugins configured for root and extensions as well.

**Comment by tatsuhiro-t** [comment]
I'll release 0.8.1 because `mvn deploy` changes signature, and nexus refused to update the existing ones.

**Comment by tatsuhiro-t** [comment]
0.8.1 was released.  release-0.8 branch contains the fixes I made to mvn deploy work for argparse4j-root and argparse4j-extensions.  It looks like `<name>` is required for argparse4j-root and argparse4j-extensions.  argparse4j-root also requires `<description>` and `<url>`.  I also added them in argparse4j-extensions, but I'm not sure they are really required.  Perhaps, only root package requires them?
